libbeat/_meta/config.reference.yml.tmpl

```diff
@@ -1110,6 +1110,13 @@ logging.files:
 # Set to true to enable the monitoring reporter.
 #monitoring.enabled: false
 
+# If output.elasticsearch is enabled, monitoring data for this {{.BeatName | title}} instance
+# will be associated with the Elasticsearch cluster referenced by output.elasticsearch in
+# the Stack Monitoring UI. However, if a different output is enabled, the monitoring data will
+# show up under a Standalone Cluster in the UI. If you want the monitoring data to be associated
+# with a specific Elasticsearch cluster in the UI, specify that cluster's UUID here.
+#monitoring.override_cluster_uuid:
```

Comment:
@dedemorton WDYT of this language?

Reply:
I wouldn't start by talking saying, "If output.elasticsearch is enabled" because skim readers might think the section doesn't apply to them. Instead, I would start with a description of what the setting contains, then I'd describe the reasons for setting it. How about something like this:

```
Sets the UUID of the Elasticsearch cluster under which monitoring data for this
{{.BeatName | title}} instance will appear in the Stack Monitoring UI. If output.elasticsearch
is enabled, the UUID is set by default. If a different output is enabled, you must specify this
setting, or monitoring data about this {{.BeatName | title}} instance will appear under a
standalone cluster.
```

Maybe a bit wordy, tho. WDYT?

Reply:
I can't recall a place in the public documentation right now where we discuss what a cluster UUID is and what its function is. I'm a bit on the fence over whether this case is common enough that we should document it more thoroughly but generally I learn toward the more documentation the better. :)

Either way, I think that we should also provide clear instructions to a user on retrieving a the cluster UUID for the cluster that they wish to associate a monitored component with.

Reply:
@cachedout In addition to @dedemorton's suggested changes, I can add a sentence about _how_ to determine the cluster UUID.

Given that this is an inline comment in a configuration file, I'm not sure there's space to talk about _what_ a cluster UUID is and what it's function is — that seems like something that belongs in our online ES docs. If you agree then perhaps file a docs issue (or PR) in the ES repo for adding the information you'd like to see?

Reply:
We can add more detail to the Beats docs where we talk about monitoring. IMO it's important to keep the description in the config file as brief as possible, or the file will become unreadable over time.

Reply:
How about naming the setting just `cluster_uuid` and saying that if the elasticsearch output is configured, then the UUID will be derived from said output if this setting is not configured.

Reply:
I wanted to add the `override_` prefix to indicate more strongly that this will be the cluster UUID that is used when it is set, regardless of any other configuration. But I'm good with your suggestion as well, @urso. I think perhaps the name is not as critical as the inline comment and web site documentation about it.

Reply:
:) Reason I ask for cluster_uuid is, cause I'd prefer it to be configured most of the time. It's the one to use for users not using the ES output always, meaning there is nothing to override. That is: we only 'derive' it from the ES output if available. In the end I don't mind as much about the exact naming, but agree the doc should be clear.

Reply:
> Checking the code how we get the UUID from the ES output I wouldn't be surprised if we publish a few events without a cluster uuid at all.

Good point. I will put up a separate PR to prevent this when `output.elasticsearch` is being used. I already did something like this for the new `beat` Metricbeat module: https://github.com/elastic/beats/pull/13020.

Reply:
Based on all the above feedback, I've updated the setting name and inline comment in 78b25d4d069f47681667b112f5d6547d09628f60. 

I tried to keep the comment short and convey that we expect users to set this setting unless they are using `output.elasticsearch`, in which case we automatically derive the cluster UUID.

Please let me know what you think.

Reply:
> I will put up a separate PR to prevent this when output.elasticsearch is being used.

PR is up: https://github.com/elastic/beats/pull/13251